### PR TITLE
fix: check workflow exists when executing

### PR DESCRIPTION
**Pull request by eloycoto**
User may introduce a typo and the output is not great:

```
2023-05-11 10:23:13.200 ERROR 52759 --- [         task-6] .a.i.SimpleAsyncUncaughtExceptionHandler : Unexpected exception occurred invoking async method: public void com.redhat.parodos.workflow.execution.service.WorkFlowExecutorImpl.executeAsync(java.util.UUID,java.lang.String,com.redhat.parodos.workflows.work.WorkContext,java.util.UUID)

java.lang.NullPointerException: Cannot invoke "com.redhat.parodos.workflow.definition.entity.WorkFlowDefinition.getName()" because "workflow" is null
        at com.redhat.parodos.workflow.execution.aspect.WorkFlowExecutionFactory.isMainWorkFlow(WorkFlowExecutionFactory.java:55) ~[classes!/:na]
        at com.redhat.parodos.workflow.execution.aspect.WorkFlowExecutionFactory.createExecutionHandler(WorkFlowExecutionFactory.java:38) ~[classes!/:na]
        at com.redhat.parodos.workflow.execution.aspect.WorkFlowExecutionAspect.executeAroundAdvice(WorkFlowExecutionAspect.java:86) ~[classes!/:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at com.redhat.parodos.workflows.workflow.ParallelFlow$$EnhancerBySpringCGLIB$$7530a326.execute(<generated>) ~[workflow-engine-1.0.11-SNAPSHOT.jar!/:na]
        at com.redhat.parodos.workflows.engine.WorkFlowEngineImpl.run(WorkFlowEngineImpl.java:38) ~[workflow-engine-1.0.11-SNAPSHOT.jar!/:na]
        at com.redhat.parodos.workflow.execution.service.WorkFlowExecutorImpl.execute(WorkFlowExecutorImpl.java:35) ~[classes!/:na]
        at com.redhat.parodos.workflow.execution.service.WorkFlowExecutorImpl.executeAsync(WorkFlowExecutorImpl.java:27) ~[classes!/:na]
        at com.redhat.parodos.workflow.execution.service.WorkFlowExecutorImpl$$FastClassBySpringCGLIB$$510b4225.invoke(<generated>) ~[classes!/:na]
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[workflow-examples-1.0.11-SNAPSHOT-jar-with-dependencies.jar:5.3.20]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115) ~[spring-aop-5.3.20.jar!/:5.3.20]
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
        at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (FLPATH-xxxx)*:
Fixes #

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [ ] Auto-generated SDK code

**Impacted services**
- [ ] Workflow Service
- [ ] Notification Service

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java [hunk 1]
```diff
@@ -82,6 +82,10 @@ public WorkReport executeAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, W
 
 		/* get workflow definition entity */
 		WorkFlowDefinition workFlowDefinition = this.workFlowDefinitionRepository.findFirstByName(workflowName);
+		if (workFlowDefinition == null) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("Cannot find workflow '"+ workflowName + "'"));
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java:86]: Can you use WorkflowDefinitionException or create
WorkflowDefinitionNotFoundException ?

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java [hunk 2]
```diff
@@ -82,6 +83,11 @@ public WorkReport executeAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, W
 
 		/* get workflow definition entity */
 		WorkFlowDefinition workFlowDefinition = this.workFlowDefinitionRepository.findFirstByName(workflowName);
+		if (workFlowDefinition == null) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new WorkflowDefinitionNotFoundException("Cannot find workflow '" + workflowName + "'"));
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java:88]: recently I made some refactor around the exceptions and we can use it here instead:
```
new ResourceNotFoundException(ResourceType.WORKFLOW_DEFINITION, IDType.NAME, workflowName)
```

Comment (eloycoto) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java]: fixed

File: workflow-service/src/test/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspectTest.java [hunk 3]
```diff
@@ -136,6 +136,26 @@ public void ExecuteAroundAdviceWithValidDataTest() {
...
+		when(workFlow.getName()).thenReturn(TEST_WORK_FLOW);
+
+		// when
+		WorkReport workReport = this.workFlowExecutionAspect.executeAroundAdvice(proceedingJoinPoint, workContext);
+
+		// then
+		assertNotNull(workReport);
+		assertEquals(workReport.getStatus().toString(), FAILED);
+		assertNotNull(workReport.getError());
+		verify(this.workFlowDefinitionRepository, times(1)).findFirstByName(TEST_WORK_FLOW);
```
Comment (masayag) [inline workflow-service/src/test/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspectTest.java:156]: pls add an assertion of the expected exception to be of type ResourceNotFoundException

**Review by masayag — APPROVED** [review]

File: workflow-service/src/test/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspectTest.java [hunk 4]
```diff
@@ -132,10 +134,34 @@ public void ExecuteAroundAdviceWithValidDataTest() {
 		assertEquals(workReport.getStatus().toString(), COMPLETED);
 		assertEquals(workReport.getWorkContext().get(WORKFLOW_DEFINITION_NAME), TEST_WORK_FLOW);
 		assertEquals(workReport.getWorkContext().get(PROJECT_ID), projectID);
+		assertThat(workReport.getError()).isInstanceOf(ResourceNotFoundException.class);
```
Comment (masayag) [inline workflow-service/src/test/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspectTest.java:137]: @eloycoto pls remove this change or replace with
```
assertThat(workReport.getError()).isNull();
```
This is a positive code path that shouldn't end with a failure

**Pull request merged**